File: backend/src/main/java/com/healthsphere/backend/service/MedicalRecordService.java
package com.healthsphere.backend.service;

import com.healthsphere.backend.dto.MedicalRecordRequest;
import com.healthsphere.backend.dto.MedicalRecordResponse;

import java.util.List;

public interface MedicalRecordService {

    MedicalRecordResponse createMedicalRecord(MedicalRecordRequest request);

    List<MedicalRecordResponse> getAllMedicalRecords();

    MedicalRecordResponse getMedicalRecordById(Long id);

    MedicalRecordResponse updateMedicalRecord(Long id,
                                              MedicalRecordRequest request);

    void deleteMedicalRecord(Long id);
}

File: frontend/src/services/medicalRecordService.js
import api from "./api";

const URL = "/medical-records";

const medicalRecordService = {

  getAllMedicalRecords: async () => {
    const response = await api.get(URL);
    return response.data;
  },

  getMedicalRecordById: async (id) => {
    const response = await api.get(`${URL}/${id}`);
    return response.data;
  },

  createMedicalRecord: async (record) => {
    const response = await api.post(URL, record);
    return response.data;
  },

  updateMedicalRecord: async (id, record) => {
    const response = await api.put(`${URL}/${id}`, record);
    return response.data;
  },

  deleteMedicalRecord: async (id) => {
    const response = await api.delete(`${URL}/${id}`);
    return response.data;
  },

};

export default medicalRecordService;